dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -917,6 +919,105 @@ public User setUserPrivilege(@Parameter(hidden = true, name = "user")@Auth User
                 .collect(Collectors.toList());
     }
 
+    @GET
+    @Timed
+    @UnitOfWork(readOnly = true)
+    @Path("/{userId}/cloudInstances")
+    @ApiOperation(value = "See OpenApi for details", hidden = true)
+    @Operation(operationId = "getUserCloudInstances", description = "Get all of cloud instances belonging to the user", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    public Set<CloudInstance> getUserCloudInstances(@Parameter(hidden = true, name = "user")@Auth User authUser,
+        @ApiParam(name = "userId", required = true, value = "User to update") @PathParam("userId") @Parameter(name = "userId", in = ParameterIn.PATH, description = "User to update", required = true) long userId) {
+        final User user = userDAO.findById(userId);
+        checkUser(authUser, userId);
+        return user.getCloudInstances();
+    }
+
+    @POST
+    @Timed
+    @UnitOfWork()
+    @Path("/{userId}/cloudInstances")
+    @Operation(operationId = "postUserCloudInstance", description = "Create a new cloud instances belonging to the user", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    @Consumes(MediaType.APPLICATION_JSON)
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = HttpStatus.SC_OK + ""),
+            @ApiResponse(responseCode = HttpStatus.SC_FORBIDDEN + ""),
+            @ApiResponse(responseCode = HttpStatus.SC_UNAUTHORIZED + "")
+    })
+    @ApiOperation(value = "See OpenApi for details", hidden = true)
+    public Set<CloudInstance> postUserCloudInstance(@Parameter(hidden = true, name = "user")@Auth User authUser,
+            @PathParam("userId") @Parameter(name = "userId", in = ParameterIn.PATH, description = "User to create cloud instance for", required = true) long userId,
+            @Parameter(description = "Cloud instance to replace for a user", name = "Cloud Instance", required = true) CloudInstance cloudInstanceBody) {
+        final User user = userDAO.findById(userId);
+        checkUser(authUser, userId);
+        CloudInstance cloudInstanceToBeAdded = new CloudInstance();
+        cloudInstanceToBeAdded.setPartner(cloudInstanceBody.getPartner());
+        cloudInstanceToBeAdded.setUrl(cloudInstanceBody.getUrl());
+        cloudInstanceToBeAdded.setSupportsFileImports(cloudInstanceBody.isSupportsFileImports());
+        cloudInstanceToBeAdded.setSupportsHttpImports(cloudInstanceBody.isSupportsHttpImports());
+        cloudInstanceToBeAdded.setSupportedLanguages(cloudInstanceBody.getSupportedLanguages());
+        // TODO: Figure how to make this not required (already adding the instance to the user)
+        cloudInstanceToBeAdded.setUser(user);
+
+        user.getCloudInstances().add(cloudInstanceToBeAdded);
+        return user.getCloudInstances();
```

Comment:
Don't you to call useDAO.update?

Reply:
elaborate?

Reply:
You're creating a cloud instance Java object. Don't you need to persist it to the database?

Reply:
tests appear to indicate otherwise